lint-tasks.yml

```diff
@@ -31,13 +42,29 @@ tasks:
         vars:
           FLAGS: "--dry-run"
 
-  cpp-fix:
-    sources: *cpp_source_files
+  cpp-format-fix:
+    sources: *cpp_format_source_files
     cmds:
       - task: "clang-format"
         vars:
           FLAGS: "-i"
 
+  cpp-static-check:
+    # Alias task to `cpp-static-fix` since we don't currently support automatic fixes.
+    # NOTE: clang-tidy does have the ability to fix some errors, but the fixes can be inaccurate.
+    # When we eventually determine which errors can be safely fixed, we'll allow clang-tidy to
+    # fix them.
+    aliases: ["cpp-static-fix"]
+    sources:
+      - "{{.G_SRC_CLP_FFI_JS_DIR}}/**/*.cpp"
+      - "{{.G_SRC_CLP_FFI_JS_DIR}}/**/*.h"
+      - "{{.G_SRC_CLP_FFI_JS_DIR}}/**/*.hpp"
```

Comment:
Do we need to include all source files? Say if some header in `clp-ffi-js`' dependency changes, it would also impact the compilation in `clp-ffi-js`' source code, right?

Reply:
It would, but this task doesn't require compiling clp-ffi-js, right? (I might be misunderstanding your question.)

Reply:
I can't come up with an exact example since I'm not too familiar with clang-tidy rules. However, from my experience working with static analysis, any dependency changes can cause the checks in our code to fail. (In fact, sometime it's easier to fix violations in the dependencies when there are too many violations due to a same issue in the dependency. )

Please correct me if this imaginary example is impossible - say we recently updated some function signature in CLP core code that some parameter changes its type from int64 to int32, but our clp-ffi-js code remains the same. If we keep passing an int64 value to the modified function, wouldn't a precision / value loss violation be raised?

Reply:
Oh I see what you're saying. Yes, we should include the dependency's source files.